src/main/java/com/cleanroommc/groovyscript/compat/mods/advancedmortars/Mortar.java

```diff
@@ -0,0 +1,153 @@
+package com.cleanroommc.groovyscript.compat.mods.advancedmortars;
+
+import com.cleanroommc.groovyscript.api.GroovyBlacklist;
+import com.cleanroommc.groovyscript.api.GroovyLog;
+import com.cleanroommc.groovyscript.api.IIngredient;
+import com.cleanroommc.groovyscript.core.mixin.advancedmortars.RegistryRecipeMortarAccessor;
+import com.cleanroommc.groovyscript.helper.ingredient.IngredientHelper;
+import com.cleanroommc.groovyscript.helper.recipe.AbstractRecipeBuilder;
+import com.cleanroommc.groovyscript.registry.VirtualizedRegistry;
+import com.codetaylor.mc.advancedmortars.modules.mortar.api.MortarAPI;
+import com.codetaylor.mc.advancedmortars.modules.mortar.recipe.RecipeMortar;
+import com.codetaylor.mc.advancedmortars.modules.mortar.reference.EnumMortarType;
+import net.minecraft.item.ItemStack;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class Mortar extends VirtualizedRegistry<RecipeMortar> {
+
+    public Mortar() {
+        super();
+    }
+
+    public RecipeBuilder recipeBuilder() {
+        return new RecipeBuilder();
+    }
+
+    @Override
+    @GroovyBlacklist
+    public void onReload() {
+        removeScripted().forEach(this::remove);
```

Comment:
remove runs `addBackup(recipe);`, so you shouldnt call it in relolad. try
```suggestion
        removeScripted().forEach(recipe -> ((RegistryRecipeMortarAccessor) MortarAPI.RECIPE_REGISTRY).getRecipeMap().values().forEach(list -> list.removeIf(r -> r == recipe)));